### PR TITLE
Localtime/daylight savings enhancements and fixes

### PR DESCRIPTION
**Pull request by MishimaHaruna**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

This adds support for choosing between localtime (default) and UTC in the script commands `getcalendartime()` and `gettime()`, through an optional argument. The default behavior when the argument is omitted is to use localtime (for backward compatibility).

A fix for a test failure in `getcalendartime()` when using localtime and daylight savings are on is included (as the function implicitly assumed that DST is always off). A similar issue has also been fixed in `clif_attendance_getendtime()`.

Bonus: constants are provided for `gettimetick()` (and a DST quirk is documented in type 1 / `GETTIMETICK_HOUROFDAY_S`) and the `gettime()` constants are moved to source.

**Issues addressed:** None that I know of

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by MrKeiKun** [comment]
Some changes here reminds me of `date.c `and `date.h` within `src/map/`.

File: npc/dev/test.txt [hunk 1]
```diff
@@ -151,20 +151,25 @@ function	script	F_TestVarOfAnotherNPC	{
...
 	// We give it a range of 70 seconds more or less in the result because:
 	// 1. getcalendartime doesn't take seconds into account, so we may have differing seconds;
 	// 2. We can't mock time, the second may change during the commands execution,
 	//    so we give it a few seconds to spare.
 	// This still gives a good test, because we are looking for 24 hours into the future,
 	// ~3 minutes aren't really an issue for correcteness here.
-	.@check = (.@result >= .@expected - 70 && .@result <= .@expected + 70);
-	
+	.@check = (.@result_fromlocal >= .@expected - 70 && .@result_fromlocal <= .@expected + 70);
+	.@check = (.@result_fromutc >= .@expected - 70 && .@result_fromutc <= .@expected + 70);
```
Comment (guilherme-gm) [inline npc/dev/test.txt:171]: Here the check on line 170 is lost as test result, isn't it?

I think it needs some sort of `&&` to check if both are passing

File: npc/dev/test.txt [hunk 2]
```diff
@@ -151,20 +151,25 @@ function	script	F_TestVarOfAnotherNPC	{
...
-	.@result = getcalendartime(.@hour, .@min);
-	.@expected = gettimetick(2) + 24 * 60 * 60;
-	
+	.@local_hour = gettime(GETTIME_HOUR, true);
+	.@local_min = gettime(GETTIME_MINUTE, true);
+	.@result_fromlocal = getcalendartime(.@hour, .@min, -1, -1, true);
+
+	.@utc_hour = gettime(GETTIME_HOUR, false);
+	.@utc_min = gettime(GETTIME_MINUTE, false);
+	.@result_fromutc = getcalendartime(.@hour, .@min, -1, -1, false);
```
Comment (guilherme-gm) [inline npc/dev/test.txt:160]: and here `.@hour` / `.@min` is no longer defined, should be `.@local_*` and `.@utc_*`

File: src/map/script.c [hunk 3]
```diff
@@ -26770,8 +26810,11 @@ static BUILDIN(mergeitem)
 	return true;
 }
 
-// getcalendartime(<day of month>, <day of week>{, <hour>{, <minute>}});
-// Returns the UNIX Timestamp of next ocurrency of given time
+/**
+ * Returns the UNIX Timestamp of next ocurrency of given time.
+ *
+ * getcalendartime(<day of month>, <day of week>{, <hour>{, <minute>{, <input_is_localtime>}}});
```
Comment (guilherme-gm) [inline src/map/script.c:26816]: oh, it looks like I wrote this comment totally wrong back then 🤦‍♂️ 

hour and minute are actually params 1 and 2. the script_commands is correct, but it is wrong here

```
getcalendartime(<hour>, <minute>{, <day of month>{, <day of week>{, <input_is_localtime>}}})
```

File: npc/dev/test.txt (hunk 1 above)
Comment (MishimaHaruna) [reply npc/dev/test.txt:171]: Oops my bad, I was originally testing this by commenting out the other one and forgot to update it when I decided to keep both

File: src/map/script.c (hunk 3 above)
Comment (MishimaHaruna) [reply src/map/script.c]: Right, I didn't notice this

**Review by guilherme-gm — APPROVED** [review]
everything is working now :D

**Comment by MishimaHaruna** [comment]
Thank you very much! I'll merge it now

**Pull request merged**